src/modules/roc_core/target_linux/roc_core/timer.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2020 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+//! @file roc_core/target_linux/roc_core/timer.h
+//! @brief Thread-safe timer.
+
+#ifndef ROC_CORE_TIMER_H_
+#define ROC_CORE_TIMER_H_
+
+#include <sys/timerfd.h>
+
+#include "roc_core/atomic.h"
+#include "roc_core/noncopyable.h"
+#include "roc_core/seqlock.h"
+#include "roc_core/time.h"
+
+namespace roc {
+namespace core {
+
+//! Thread-safe timer.
+class Timer : public NonCopyable<> {
+public:
+    Timer();
+
+    virtual ~Timer();
+
+    //! Set timer deadline.
+    //! Can be called concurrently, but only one concurrent call will succeed.
+    //! Returns false if the call failed because of another concurrent call.
+    //! Is lock-free if Semaphore::post() is so (which is true of modern plarforms).
```

Comment:
The note on Semaphore::post() in not relevant here.

Reply:
ok